### PR TITLE
Fix Product Update View

### PR DESCRIPTION
**Pull request by sachintom999**
## Background

 - The Product Update View is currently not working
 
### Issues
- It does not allows to save the changes to an existing product without updating the product name. ( due to checking slug )
- if we tries to save by updating the product name, it errors out with null constraint error for "object_id" field
- It is demonstrated as below -

before-fix

### Solution
- the first issue can be solved by skipping the slug checking in the `clean_name()` method in the `ProductForm`
- the fix of the second issue ( updating the Product ) has a lot of moving parts:
     - introduce the `clean()` method in `ProductForm`
     - move the `owner` validation from `CreateProductView` to this `clean` method
     - allowing user to select either `organisation` or the `user` itself as the owner
     - render organisation/user value in the update view template from the db
     - throw an error if user selects both organisation and himself OR don't select any of the two
     - render the above errors in the template

### Changes
- `product_management/views.py`
- `product_management/forms.py`
- `product_management/templates/product_management/forms/product_form.html`

### Testing 
- After the changes the testing has been done for the below scenarios
    - Trying to create a new product 
        - without any owner  -> resulting in error :  `You have to select an owner`
        - with both `organisation` and `user` both selected -> resulting in error : `A product cannot be owned by a person and an organisation`
    - Successfully creating a new product
    - Updating the newly created product , without updating product name ( solving the slug checking issue)
    - Updating the owner to  a) an organisation  b) user.  Both rendering correctly in the template after the update
    - In the update view, saving the product without making any changes to the existing field
    - Please find the demo below : 
    
after-fix

### CODE REVIEW

File: product_management/forms.py [hunk 1]
```diff
@@ -63,7 +64,7 @@ class ProductForm(forms.ModelForm):
     organisation = forms.ModelChoiceField(
         empty_label="Select an organisation",
         required=False,
```
Comment (dogukanteber) [inline product_management/forms.py:66]: Can you add the following here:

`TODO: We should not get all the organizations here. After the organization hierarchy is constructed, we need to filter based on the current user's organizations.`

**Review by dogukanteber — CHANGES_REQUESTED** [review]
Perfect job @sachintom999. I have been meaning to make this refactoring but did not know updating is not working. Thank you very much for your contribution.

There are two super minor modifications I want you to make. After that, we can merge it.

File: product_management/templates/product_management/forms/product_form.html [hunk 2]
```diff
@@ -33,8 +33,18 @@ <h2 class="text-base font-semibold leading-7 text-gray-900">Product Information<
...
                                 Don't have an organisation? <a class="font-semibold"
-                                    href="{{ url('create-organisation') }}">Create one.</a>
+                                href="{{ url('create-organisation') }}">Create one.</a>
                             </p>
+                            
+                            <div class="my-2">
+                                {% if form.organisation.errors %}
+                            <ul class="errorlist">
+                                {% for error in form.organisation.errors %}
+                                <li class="text-xs italic text-red-600 ">Error : {{ error }}</li>
```
Comment (dogukanteber) [inline product_management/templates/product_management/forms/product_form.html:43]: We can also remove the italic class from the class list and  `Error : ` string from the text.

**Comment by sachintom999** [comment]
> Perfect job @sachintom999. I have been meaning to make this refactoring but did not know updating is not working. Thank you very much for your contribution.
> 
> There are two super minor modifications I want you to make. After that, we can merge it.

hi @dogukanteber,
Thanks for reviewing 🙂 . I have made the requested changes and pushed.

**Review by dogukanteber — APPROVED** [review]

**Comment by dogukanteber** [comment]
Awesome. Let's merge it 🚀

**Comment by adrianmcphee** [comment]
@sachintom999  are you available to help further on OpenUnited? If so please DM me via our Discord server https://discord.com/invite/T3xevYvWey thanks

**Pull request merged**